src/visualize/AverageHandler.cpp

```diff
@@ -22,9 +23,16 @@ void AverageHandler::WriteDofValues(const CellInterface& cell, const DofType dof
                                     UnstructuredGrid* grid)
 {
     grid->DefinePointData(dof.GetName());
+    bool as3d = dof.GetNum() == 2; // allows _warp by vector for 2d displacements
     for (size_t iCorner = 0; iCorner < mGeometry.cornerCoordinates.size(); ++iCorner)
-        grid->SetPointData(pointIds[iCorner], dof.GetName(),
-                           cell.Interpolate(mGeometry.cornerCoordinates[iCorner], dof));
+    {
+        auto dofValues = cell.Interpolate(mGeometry.cornerCoordinates[iCorner], dof);
+
+        if (as3d)
+            dofValues = EigenCompanion::To3D(dofValues);
+
+        grid->SetPointData(pointIds[iCorner], dof.GetName(), dofValues);
+    }
```

Comment:
Seems like this does not depend on the handler. Maybe move to `UnstructuredGrid`?